[tests] Extract support for running an aspire app for tests

radical opened the pull request:
.. to `tests/Shared/WorkloadTesting/AspireProject.cs`, so it can be used
by other tests too.

This lays the ground work for upcoming workload tests.

radical committed:
[tests] Extract support for running an aspire app for tests

.. to `tests/Shared/WorkloadTesting/AspireProject.cs`, so it can be used
by other tests too.

eerhardt commented on tests/Shared/WorkloadTesting/AspireProject.cs:
```diff
@@ -0,0 +1,295 @@
+        testOutput.WriteLine("--------------------------- Docker info (end) ---------------------------");
+    }
+
+    public async Task DumpComponentLogsAsync(string component, ITestOutputHelper? testOutputArg = null)
```
Is this duplicated with `IntegrationServicesFixture.DumpComponentLogsAsync`?

eerhardt commented on tests/Shared/WorkloadTesting/AspireProject.cs:
```diff
@@ -0,0 +1,295 @@
+        await Process.WaitForExitAsync();
+    }
+
+    public async Task DumpDockerInfoAsync(ITestOutputHelper? testOutputArg = null)
```
Is this duplicated with `IntegrationServicesFixture.DumpDockerInfoAsync`?

eerhardt commented on tests/Shared/WorkloadTesting/AspireProject.cs:
```diff
@@ -0,0 +1,295 @@
+    public static string GetNuGetConfigPathFor(string targetFramework) =>
+        Path.Combine(BuildEnvironment.TestDataPath, "nuget8.config");
+
+    public Process? Process { get; private set; }
```
Is there really only one `Process` for an "AspireProject"? Should this be `AppHostProcess`?

radical replied:
hm good point. Right now we are tracking the only process that we start, but we could be tracking the others too. I'll fix.

eerhardt reviewed (approved):
I just had a handful of questions/thoughts. This looks good. Feel free to address the questions as you see fit.

radical commented on tests/Shared/WorkloadTesting/AspireProject.cs:
```diff
@@ -0,0 +1,295 @@
+{
+    public static Lazy<HttpClient> Client => new(CreateHttpClient);
+    public static string GetNuGetConfigPathFor(string targetFramework) =>
+        Path.Combine(BuildEnvironment.TestDataPath, "nuget8.config");
```
Oops, this will be part of the workloads PR, can be removed from here.

radical committed:
Merge remote-tracking branch 'origin/main' into e2e-fixture

radical committed:
Address review feedback from @ eerhardt

- Rename `AspireProject.Process` to `AspireProject.AppHostProcess`
- remove nuget8.config which isn't needed yet

radical committed:
Update tests/Aspire.EndToEnd.Tests/IntegrationServicesTests.cs

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

radical commented on tests/Aspire.EndToEnd.Tests/IntegrationServicesFixture.cs:
```diff
@@ -58,211 +49,33 @@ public IntegrationServicesFixture(IMessageSink diagnosticMessageSink)
-                            msg += $" Content:{Environment.NewLine}{contentStr}";
-                        }
-
-                        _testOutput.WriteLine(msg);
```
great catch, thank you!

radical committed:
address review feedback from @ eerhardt

radical committed:
Address review feedback from @ eerhardt, and remove some duplication

radical commented:
Test failures are unrelated to the changes here - https://github.com/dotnet/aspire/issues/3609 .

radical commented:
/backport to release/8.0
